### PR TITLE
Add support for nodelocal DNS, update Canal/Calico templates with privileged and add rke-tools:v0.1.53

### PR DESCRIPTION
**Pull request by superseb**
https://github.com/rancher/rke/issues/1050

Depends on:
https://github.com/rancher/kontainer-driver-metadata/pull/149 && https://github.com/rancher/kontainer-driver-metadata/pull/150
https://github.com/rancher/types/pull/1108 && https://github.com/rancher/types/pull/1114

### CODE REVIEW

File: cluster/validation.go [hunk 1]
```diff
@@ -344,6 +344,11 @@ func validateDNSImages(c *Cluster) error {
 			return fmt.Errorf("coredns autoscaler image is not populated")
 		}
 	}
+	if c.DNS.Nodelocal != nil {
+		if len(c.SystemImages.Nodelocal) == 0 {
```
Comment (luthermonson) [inline cluster/validation.go]: do this in one line?

File: cluster/validation.go [hunk 2]
```diff
@@ -344,6 +344,11 @@ func validateDNSImages(c *Cluster) error {
...
+			return fmt.Errorf("nodelocal image is not populated")
```
Comment (luthermonson) [inline cluster/validation.go]: error.New

Comment (superseb) [reply cluster/validation.go]: Replaced all non formatted errors to `errors.New`

File: cluster/validation.go (hunk 1 above)
Comment (superseb) [reply cluster/validation.go]: This was to keep the style with the other systemimages checks, but as this is only one separate condition I merged them

**Pull request merged**